indexing-service/src/main/java/org/apache/druid/indexing/compact/MSQCompactionJobTemplate.java

```diff
@@ -0,0 +1,191 @@
+/*
```

Comment:
when I was playing with this on my local I accidentally got into a state where I had defined an expected granularity of MONTH, but the query in the template was DAY. It looked like it just went into an infinite compact loop. Could we force the templated query to honor segment granularity from the state matcher by injecting it? I guess this forces everyone creating a rule to select a segment gran though, which is probably not desired.